### PR TITLE
fix(curriculum): update tests to better catch common errors

### LINKED ISSUE #55597: Cat photo app - step 56 example should be updated?
Since step 56 of the cat photo app continues to cause confusion for campers, and the most common reason cited for this is that the example code shown shows the text on the opposite side of the input compared to how the actual app code is written, why don't we consider just removing the example code entirely?

If people already know what an attribute is, then adding the for attribute should be easy to do. They don't need the example?

If they don't know how to do it, then they will ask on the forum which is what everyone does for this step anyway. We may lessen the total number of questions if we just erase the sample code.

Just a thought.

### PR DESCRIPTION
**Pull request by lasjorg**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55597

Refactored the tests to help catch some of the most common errors. I went through a handful of forum threads, there are more edge cases, but we can't catch them all unless we add more regex tests. But if there are any more obvious errors we should test for, let me know.

<s>This should also allow us to keep the example code as is - see related issue.</s>

Edit: updated the example code as well.

### CODE REVIEW

**Comment by lasjorg** [comment]
I somehow managed to rename my main branch to this branch and push it. Not really sure if I should redo this PR again to avoid any issues?

**Comment by moT01** [comment]
The main on your fork looks like it's in sync with fcc/main. So I think you might have just made a new branch and pushed that? That's what it appears like anyway - this branch is on your fork.

**Comment by lasjorg** [comment]
@moT01

I think I did this (I really can't remember now):

Rename the branch back to main.

Create it again from main.

Update main by fetching from upstream and pushing to origin.

But then I couldn't see the changes I made locally on the branch, so I fetched from my remote branch back down locally and did a reset.

```
git fetch origin fix/cat-photo-app-update-tests
git reset --hard origin/fix/cat-photo-app-update-tests
```

Now I can see the changes again. I think the history should be OK (did a `log --oneline`)

---

As an aside, I finally updated my WSL `.bashrc` to show the branch I'm on.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efc4f528d6a74d05e68af74.md [hunk 1]
```diff
@@ -9,73 +9,90 @@ dashedName: step-56
...
+You should add a new `label` element after the checkbox element. Make sure it has both an opening and closing tag.
 
 ```js
-const checkboxInputElem = $('input[type="checkbox"]')[0];
-assert(
-  /Loving/i.test(checkboxInputElem.nextElementSibling.innerText.replace(/\s+/g, ' '))
+assert.isTrue(
+  document.querySelectorAll('label').length === 3 &&
+    code.match(/<\/label\>/g).length === 3
 );
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efc4f528d6a74d05e68af74.md:48]:
```suggestion
assert.lengthOf(document.querySelectorAll('label'), 3);
assert.lengthOf(code.match(/<\/label\>/g), 3);
```

Nit to use more specific assertion methods.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efc4f528d6a74d05e68af74.md [hunk 2]
```diff
@@ -9,73 +9,90 @@ dashedName: step-56
...
 Associate the text `Loving` with the checkbox by nesting only the text `Loving` in a `label` element and giving it an appropriate `for` attribute.
 
 # --hints--
 
-You should make sure the checkbox is still present.
+You should make sure the checkbox element is still present, and you did not make any changes to it. It should stay as it was in the starting code.
 
 ```js
-assert($('input[type="checkbox"]')[0]);
+const checkboxElementRegex = /\s*<input\s+id\s*=\s*"loving"\s*type\s*=\s*"checkbox"\s*>/;
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efc4f528d6a74d05e68af74.md:31]: Do we need to revert to a regex?

I understand the goal is to keep campers from editing the input element, but if they add spacing do we really need to care? As long as the attributes and DOM layout are what we want?

Comment (lasjorg) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efc4f528d6a74d05e68af74.md:31]: The problem was that some changes to the input element, caused us to assert the wrong thing.

```
<input id="loving" type="checkbox" <label>Loving</label>
```
> The text Loving should be wrapped in a label element.

With the regex it is this instead:

> You should make sure the checkbox element is still present, and you did not make any changes to it. It should stay as it was in the starting code.

**Review by naomi-lgbt — APPROVED** [review]

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efc4f528d6a74d05e68af74.md (hunk 2 above)
Comment (naomi-lgbt) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efc4f528d6a74d05e68af74.md:31]: Gosh darn browsers and their ability to parse malformed code 🤣

**Review by jeremylt — APPROVED** [review]

**Pull request merged**